src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -967,6 +979,12 @@ private Index createInstance(IndexMetadata indexDef)
             }
             catch (Exception e)
             {
+                if (e instanceof ConfigurationException && INDEX_UNKNOWN_IGNORE.getBoolean())
```

Comment:
This would also catch any `ConfigurationException` in the constructor of the index, whereas the error message is only about getting the class. I think we should either use separate try-catch blocks or use a more specific exception.

Reply:
I don't think this is resolved. The try-catch block would still enter into the unknown class path if the constructor of a known index throws any other equally generic `ConfigurationException` for some other reason.

Reply:
eh, `ConfigurationException` is unchecked 😔

Reply:
I change my mind here and removing that change,  as it isn't needed. (And I would like changes as minimal as possible for the sake of future rebases.)

The only place ConfigurationException can be thrown is in `FBUtilities.classForName(…)`

As `newInstance(…)` will always wrap any throwable inside a `InvocationTargetException`. 
(e.g. see `DirectConstructorHandleAccessor.newInstance(…)`)

I'll add a comment in the code to make this clear.

Reply:
Agree, the wrapping within `InvocationTargetException` should make it work, thanks for adding the comment.